edx_sga/sga.py

```diff
@@ -745,28 +745,33 @@ def prepare_download_submissions(self, request, suffix=''):  # pylint: disable=u
         user = self.get_real_user()
         require(user)
         zip_file_ready = False
+        unicode_block_id = unicode(self.block_id)
+        unicode_course_id = unicode(self.block_course_id)
 
-        if self.is_zip_file_available():
+        if self.is_zip_file_available(user):
```

Comment:
Look [here](https://github.com/mitodl/edx-sga/blob/004ee40bb9f9fff43c4cda90f6f5d17e8a85db22/edx_sga/sga.py#L745) and [here](https://github.com/mitodl/edx-sga/blob/004ee40bb9f9fff43c4cda90f6f5d17e8a85db22/edx_sga/sga.py#L751). If `is_zip_file_available` is responsible for extracting the user out of the current context, then we are doing that work twice within `prepare_download_submissions`. 

`is_zip_file_available` is simply a utility function. It doesn't make any sense to make it aware of the runtime and give it the responsibility of fetching the actual user object. It makes perfect sense for our view functions (the ones decorated with `@XBlock.handler`) to do that work, because those functions handle HTTP requests and they are the entry point to our application logic. Also, if you look [here](https://github.com/mitodl/edx-sga/blob/004ee40bb9f9fff43c4cda90f6f5d17e8a85db22/edx_sga/sga.py#L790), that handler function did not specify `require(user)`, which obscured the fact that it actually does require a logged-in user to work.